bundles/org.openhab.binding.modbus.sunspec/src/main/java/org/openhab/binding/modbus/sunspec/internal/discovery/SunspecDiscoveryProcess.java

```diff
@@ -282,8 +283,11 @@ private void createDiscoveryResult(ModelBlock block) {
             return;
         }
 
-        ThingUID thingUID = new ThingUID(SUPPORTED_THING_TYPES_UIDS.get(block.moduleID), handler.getUID(),
-                Integer.toString(block.address));
+        ThingTypeUID thingTypeUID = SUPPORTED_THING_TYPES_UIDS.get(block.moduleID);
+        if (thingTypeUID == null) {
+            logger.warn("Found model block but no corresponding thing type UID present: {}", block.moduleID);
+        }
+        ThingUID thingUID = new ThingUID(thingTypeUID, handler.getUID(), Integer.toString(block.address));
```

Comment:
Should you return if you find a null `thingTypeUID`? Is it ok to create a ThingUID with a null `thingTypeUID`?